### ISSUE TITLE
Hover over variable obtained via `__getattr__` should not say it is a `function`

### ISSUE THREAD

**Issue by debonte** [microsoft/pylance-release#4809]
Found while testing #4764 and #4660.

## Environment data

-   Language Server version: 2023.9.11
-   OS and version: Windows 11
-   Python version: 3.11.4

## Code Snippet

```python
from typing import Any, Literal, overload


class A:
    @overload
    def __getattr__(self, item: Literal['foo', 'bar']) -> int:
        pass

    @overload
    def __getattr__(self, item: Literal['baz']) -> str:
        pass

    @overload
    def __getattr__(self, item: str) -> int | str:
        pass

    def __getattr__(self, item: str) -> Any:
        return None

a = A()
a.bar

class B:
    bar: int
    
b = B()
b.bar
```

## Repro Steps

1. Add code above
2. Hover over `bar` of `a.bar`

## Expected behavior

Hover says that `a.bar` is a variable as it does for `b.bar`:
![image](https://github.com/microsoft/pylance-release/assets/12820898/3f76f794-b4fe-4685-9ec7-6d72b0ac40c9)

## Actual behavior

Hover says that `a.bar` is a function, I guess because `__getattr__` is a function.
![image](https://github.com/microsoft/pylance-release/assets/12820898/5d1ef7b3-5282-4a1c-95be-a15e37acba43)

**Comment by erictraut** [comment]
I agree that `(function)` is misleading, but is `(variable)` less misleading? There is no static declaration (variable or otherwise) associated with this symbol.

I don't think there's a "correct" answer here, so perhaps `(variable)` is the least surprising for users. Other options are: 1) omit the declaration type (the part in parens) in this case, 2) emit something like `(dynamic attribute)` or `(attribute)` to indicate that it isn't associated with a static declaration but is an attribute that comes from a runtime code path.

**Comment by bschnurr** [comment]
This issue has been fixed in prerelease version 2026.1.100, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20261100-27-February-2026-PreRelease)
